thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/components/MapePercentageChangeModelEvaluator.java

```diff
@@ -0,0 +1,102 @@
+ *  * KIND, either express or implied.  See the License for the
+ *  * specific language governing permissions and limitations
+ *  * under the License.
+ *
+ *
+ */
+
+package org.apache.pinot.thirdeye.detection.components;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.pinot.thirdeye.datalayer.dto.EvaluationDTO;
+import org.apache.pinot.thirdeye.datalayer.pojo.EvaluationBean;
+import org.apache.pinot.thirdeye.detection.InputDataFetcher;
+import org.apache.pinot.thirdeye.detection.spec.MapePercentageChangeModelEvaluatorSpec;
+import org.apache.pinot.thirdeye.detection.spi.components.ModelEvaluator;
+import org.apache.pinot.thirdeye.detection.spi.model.EvaluationSlice;
+import org.apache.pinot.thirdeye.detection.spi.model.InputDataSpec;
+import org.apache.pinot.thirdeye.detection.spi.model.ModelEvaluationResult;
+import org.apache.pinot.thirdeye.detection.spi.model.ModelStatus;
+import org.joda.time.Instant;
+
+
+/**
+ *  Monitor the recent mean MAPE in last 7 days, and compare that with the mean MAPE for the last 30 days.
+ *  If the percentage change dropped to a certain threshold for a metric urn, return a bad model status to trigger
+ *  auto configuration.
+ */
+public class MapePercentageChangeModelEvaluator implements ModelEvaluator<MapePercentageChangeModelEvaluatorSpec> {
+  private InputDataFetcher dataFetcher;
+  private double threshold;
+
+  @Override
+  public ModelEvaluationResult evaluateModel(Instant evaluationTimeStamp) {
+    EvaluationSlice evaluationSlice =
+        new EvaluationSlice().withStartTime(evaluationTimeStamp.toDateTime().minusDays(30).getMillis())
+            .withEndTime(evaluationTimeStamp.getMillis());
+    // fetch evaluations
+    Collection<EvaluationDTO> evaluations =
+        this.dataFetcher.fetchData(new InputDataSpec().withEvaluationSlices(Collections.singleton(evaluationSlice)))
+            .getEvaluations()
+            .get(evaluationSlice);
+    // calculate past 7 day mean MAPE for each metric urn
+    Map<String, Double> sevenDaysMeanMapeForMetricUrns =
+        getMeanMapeForEachMetricUrn(evaluations, evaluationTimeStamp, 7);
+
+    // calculate past 30 day mean MAPE for each metric urn
+    Map<String, Double> thirtyDaysMeanMapeForMetricUrns =
+        getMeanMapeForEachMetricUrn(evaluations, evaluationTimeStamp, 30);
+
+    // evaluate for each metric urn
+    Map<String, Boolean> evaluationResultForMetricUrns = sevenDaysMeanMapeForMetricUrns.entrySet()
+        .stream()
+        .collect(Collectors.toMap(Map.Entry::getKey,
+            // compare the MAPE percentage change to threshold
+            sevenDayMeanMape -> sevenDayMeanMape.getValue() / thirtyDaysMeanMapeForMetricUrns.get(sevenDayMeanMape.getKey()) - 1 <= threshold));
+
+    if (evaluationResultForMetricUrns.values().stream().allMatch(result -> result)) {
+      // if all metric urn's status is good, return overall good status
```

Comment:
This condition seems too restrict for multidimensional metrics. 
Let's keep an eye on it.

Reply:
Yeah. In the next PR, I'll add a filed to record the last tuning time.